bulk_site_creator/views.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from ast import literal_eval
+from uuid import uuid4
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect, render
+from django.urls import reverse
+from django.views.decorators.http import require_http_methods
+from django_auth_lti import const
+from django_auth_lti.decorators import lti_role_required
+from icommons_common.canvas_utils import (SessionInactivityExpirationRC) # TODO replace this
+from lti_permissions.decorators import lti_permission_required
+from common.utils import (
+    get_term_data_for_school,
+    get_canvas_site_templates_for_school,
+)
+from icommons_common.models import School, CourseInstance
+
+logger = logging.getLogger(__name__)
+
+SDK_CONTEXT = SessionInactivityExpirationRC(**settings.CANVAS_SDK_SETTINGS)
+
+
+@login_required
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Your code allows safe HTTP methods (i.e `GET`, `HEAD`, `OPTIONS`) and unsafe HTTP methods (i.e `POST`, `PUT`, `DELETE`) to perform an operation on a web application. This could impact its security.  We recommend that you to use safe http methods while performing read-only operations. If you have to use unsafe HTTP methods, use them only to change the state of an appilication. [Learn more](https://cwe.mitre.org/data/definitions/352.html).